python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,531 @@
+        new_item._description = item_dict["description"]
+
+        for child_dict in item_dict["children"]:
+            new_item._children.append(
+                PublishItem.from_dict(child_dict, parent=new_item)
+            )
+
+        new_item._global_properties = PublishData(
+            **item_dict["global_properties"])
+
+        for (k, prop_dict) in item_dict["local_properties"].iteritems():
+            new_item._local_properties[k] = PublishData(**prop_dict)
+
+        new_item._parent = parent
+
+        new_item._persistent = item_dict["persistent"]
+
+        if item_dict["context"]:
+            new_item._context = sgtk.Context(
+                sgtk.platform.current_bundle().sgtk,
+                **item_dict["context"]
+            )
+
+        for task_dict in item_dict["tasks"]:
+            new_item._tasks.append(
+                PublishTask.from_dict(task_dict, item=new_item)
+            )
+
+        return new_item
+
+    def __init__(self, name, type_spec, type_display, parent=None):
+        """
+        Initialize an item in the tree.
+
+        :param name: The display name of the item to create.
+        :param type_spec: The type specification for this item.
+        :param type_display: The type display string.
+
+        NOTE: Items store an explicit state that may not represent its actual
+        state within the overall publish tree. For example, if no ``context``
+        has been set on the item, it will be inherited from its parent in the
+        publish tree.
+        """
+
+        # NOTE: since this object is serializable, any members added should be
+        # simple python types or serializable objects.
+
+        # These members are common to all tree items.
+        self._active = True
+        self._children = []
+        self._context = None
+        self._description = None
+        self._global_properties = PublishData()
+        self._local_properties = defaultdict(PublishData)
+        self._name = name
+        self._parent = parent
+        self._persistent = False
+        self._tasks = []
+        self._type_display = type_display
+        self._type_spec = type_spec
+
+    def __iter__(self):
+        """Iterates over the item's descendents, depth first."""
+
+        for item in self._traverse_item(self):
+            yield item
+
+    def to_dict(self):
+        """
+        Returns a dictionary representation of the publish item. Typically used
+        during serialization.
+        """
+
+        converted_local_properties = {}
+        for (k, prop) in self._local_properties.iteritems():
+            converted_local_properties[k] = prop.to_dict()
+
+        context_value = None
+        # check _context here to avoid traversing parent. if no context manually
+        # assigned to the item, it will inherit it from the parent or current
+        # bundle context on deserialize.
+        if self._context:
+            context_value = {
```

Comment:
Yeah, I think it would be pretty useful in general to have a way to get a context as a dict via the core API. I've created a ticket for it and added a TODO here.